### ISSUE TITLE
[OpInfo] Ternary reference inputs missing important metadata

### ISSUE THREAD

**Issue by OmerZKol** [pytorch/pytorch#189544]
### 🐛 Describe the bug

`reference_inputs_elementwise_ternary` generates reference `SampleInput`s for `addcdiv`, `addcmul`, and `clamp`. These generated samples have two problems:
1. Generated samples have their `SampleInput.broadcasts_input` at its default value of `False`, even when it should not.
2. For `clamp`, broadcasted tensor bounds are generated with `requires_grad=True`, but backward fails when the gradient must be reduced to the original bound shape.

This is a test metadata issue, not an operator correctness issue. This metadata is used to avoid running in-place variants when broadcasting would require resizing the first input. Because the metadata is incorrect, the vmap exhaustive tests fail with a broadcast-shape error, which is why they have been added as [xfails](https://github.com/pytorch/pytorch/blob/main/test/functorch/test_vmap.py#L4488).

### Versions

PyTorch version: 2.14.0a0+gitb9655f2
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (aarch64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: version 3.31.6
Libc version: glibc-2.39

Python version: 3.12.3 (main, Jun 19 2026, 12:46:00) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.8.0-124-generic-aarch64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Architecture:                            aarch64
CPU op-mode(s):                          32-bit, 64-bit
Byte Order:                              Little Endian
CPU(s):                                  19
On-line CPU(s) list:                     0-18
Vendor ID:                               ARM
Model name:                              Neoverse-N1
Model:                                   1
Thread(s) per core:                      1
Core(s) per socket:                      19
Socket(s):                               1
Stepping:                                r3p1
BogoMIPS:                                50.00
Flags:                                   fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-18
Vulnerability Gather data sampling:      Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; CSV2, BHB
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Not affected

Versions of relevant libraries:
[pip3] mypy==1.16.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.26.4
[pip3] nvidia-cuda-cupti==13.3.75
[pip3] onnx==1.21.0
[pip3] onnx-ir==0.1.16
[pip3] onnxscript==0.6.2
[pip3] optree==0.13.0
[pip3] torch==2.14.0a0+gitb9655f2
[conda] Could not collect

cc @mruberry @zou3519 @Chillee @samdow @kshitij12345
